Utilities/MachineInventory.cs

```diff
@@ -0,0 +1,245 @@
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+namespace FortressCraft.Community.Utilities
+{
+    /// <summary>
+    ///     Generic machine inventory class with list inventory support
+    /// </summary>
+    public class MachineInventory
```

Comment:
I've made several fixes to this locally since I've started implementing
it.  I'll take a look at what other areas I need to cover but I believe I
have null checks in most places that would cause a problem.

On Wed, Jun 15, 2016 at 7:01 PM, Vici37 <notifications@github.com> wrote:

> In Utilities/MachineInventory.cs
> <https://github.com/Vici37/FCECommunityTools/pull/8#discussion_r67277299>:
>
> > @@ -0,0 +1,245 @@
> > +using System.Collections.Generic;
> > +using System.IO;
> > +using UnityEngine;
> > +
> > +namespace FortressCraft.Community.Utilities
> > +{
> > +    /// <summary>
> > +    ///     Generic machine inventory class with list inventory support
> > +    /// </summary>
> > +    public class MachineInventory
>
> Class looks helpful - is this supposed to act as the delegate for machines
> that have inventory? I would suggest making null checks in all of your
> functions for safety.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Vici37/FCECommunityTools/pull/8/files/c240e70fc17db4d3b4f020497b8fc9232266324e#r67277299>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ASi5i41XdFdTxg8z1FdjTtFsx85FHPj5ks5qMK5mgaJpZM4I0Bzh>
> .
>